test/dialect/postgresql/test_query.py

```diff
@@ -1640,6 +1641,15 @@ def test_with_ordinality_star(self, connection):
 
         eq_(connection.execute(stmt).all(), [(4, 1), (3, 2), (2, 3), (1, 4)])
 
+    def test_array_empty_with_type(self, connection):
+        stmt = select(postgresql.array([], type_=Integer))
+        eq_(connection.execute(stmt).all(), [([],)])
+
+    def test_array_empty_no_type(self, connection):
+        stmt = select(postgresql.array([]))
+        with expect_raises(exc.ProgrammingError):
+            connection.execute(stmt)
```

Comment:
since drivers are not very consistent about what error they will raise, let's move this to the test compiler just asserting that we can render the sql for an empty array without type